<a href="https://colab.research.google.com/github/sharmilamaryselvam/Google-Play-Store-App-Analysis-Dashboard/blob/main/Play_Staore_app_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Import required Libraries**

In [123]:
import numpy as np
import pandas as pd

## **Load and Read the Data**

In [124]:
country_app_data = pd.read_csv('/content/app_country_stats.csv', low_memory= False)

In [125]:
country_app_data

,id,app_id,country,installs,min_installs,score,ratings,reviews,price,free
0,1,com.picsart.studio,us,"1,000,000,000+",1.000000e+09,4.032339,12207068.0,445242.0,0.0,True
1,2,com.picsart.studio,in,"1,000,000,000+",1.000000e+09,4.232780,12207088.0,742770.0,0.0,True
2,3,com.picsart.studio,br,"1,000,000,000+",1.000000e+09,4.092960,12207068.0,435529.0,0.0,True
3,4,com.picsart.studio,id,"1,000,000,000+",1.000000e+09,4.119907,12207067.0,441799.0,0.0,True
4,5,com.picsart.studio,mx,"1,000,000,000+",1.000000e+09,4.235917,12207068.0,196689.0,0.0,True
...,...,...,...,...,...,...,...,...,...,...
111302,111335,com.tiktok.lite.go,gb,"10,000,000+",1.000000e+07,4.220000,42349.0,16.0,0.0,True
111303,111336,com.tiktok.lite.go,de,"10,000,000+",1.000000e+07,4.432692,42349.0,48.0,0.0,True
111304,111337,com.tiktok.lite.go,jp,"10,000,000+",1.000000e+07,4.116279,42349.0,17.0,0.0,True
111305,111338,com.tiktok.lite.go,kr,"10,000,000+",1.000000e+07,0.000000,0.0,0.0,0.0,True


## **Data Inspection**

In [126]:
country_app_data.shape

(111307, 10)

In [127]:
country_app_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 111307 entries, 0 to 111306
Data columns (total 10 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   id            111307 non-null  int64  
 1   app_id        111307 non-null  object 
 2   country       111307 non-null  object 
 3   installs      110988 non-null  object 
 4   min_installs  110988 non-null  float64
 5   score         73811 non-null   float64
 6   ratings       73811 non-null   float64
 7   reviews       73811 non-null   float64
 8   price         110988 non-null  float64
 9   free          111306 non-null  object 
dtypes: float64(5), int64(1), object(4)
memory usage: 8.5+ MB


## **Data Manipulation**

Since we have app_id, remove id which is as same purpose as app_id.

In [128]:
country_app_data = country_app_data.drop('id',axis = 1)

If we notice *'installs'*, it has object data type and the value is with '+' which denotes more than.

> 100000000+  = more than 1 million

So, it has the certaim no of installs without +. Remove + and convert it into numeric data type

In [129]:
country_app_data['installs'] = pd.to_numeric(country_app_data['installs']
                                .astype(str)
                                .str.replace(',' , '', regex = False)
                                .str.replace('+' , '', regex = False),
                                             errors= 'coerce')

$f$$r$$e$$e$ is in *object* data type with $T$$r$$u$$e$ & $F$$a$$l$$s$$e$. COnvert it into bool which is appropriate.

In [130]:
country_app_data['free'] =country_app_data['free'].astype(bool)

In $c$$o$$u$$n$$t$$r$$y$ column it is showing countries abbrevations. Change abbrevation into full country name to make identify the countries easier.

In [131]:
country_app_data['country'].unique()

array(['us', 'in', 'br', 'id', 'mx', 'gb', 'de', 'jp', 'kr', 'ph'],
      dtype=object)

In [132]:
country_app_data['country'].replace({'us': 'USA',
                                     'in' : 'India',
                                     'br' : 'Brazil',
                                     'id' : 'Indonesia',
                                     'mx' : 'Mexico',
                                     'gb' : 'United Kingdom',
                                     'de' : 'Germany',
                                     'jp' : 'Japan',
                                     'kr' : 'South Korea',
                                     'ph' : 'Philippines'},
                                    inplace = True)

/tmp/ipykernel_2878/4170185603.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  country_app_data['country'].replace({'us': 'USA',


In [133]:
country_app_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 111307 entries, 0 to 111306
Data columns (total 9 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   app_id        111307 non-null  object 
 1   country       111307 non-null  object 
 2   installs      110988 non-null  float64
 3   min_installs  110988 non-null  float64
 4   score         73811 non-null   float64
 5   ratings       73811 non-null   float64
 6   reviews       73811 non-null   float64
 7   price         110988 non-null  float64
 8   free          111307 non-null  bool   
dtypes: bool(1), float64(6), object(2)
memory usage: 6.9+ MB


In [134]:
country_app_data.head()

,app_id,country,installs,min_installs,score,ratings,reviews,price,free
0,com.picsart.studio,USA,1.000000e+09,1.000000e+09,4.032339,12207068.0,445242.0,0.0,True
1,com.picsart.studio,India,1.000000e+09,1.000000e+09,4.232780,12207088.0,742770.0,0.0,True
2,com.picsart.studio,Brazil,1.000000e+09,1.000000e+09,4.092960,12207068.0,435529.0,0.0,True
3,com.picsart.studio,Indonesia,1.000000e+09,1.000000e+09,4.119907,12207067.0,441799.0,0.0,True
4,com.picsart.studio,Mexico,1.000000e+09,1.000000e+09,4.235917,12207068.0,196689.0,0.0,True


## **Data Cleaning**

In [135]:
country_app_data.isnull().sum()

,0
app_id,0
country,0
installs,319
min_installs,319
score,37496
ratings,37496
reviews,37496
price,319
free,0


In [136]:
install_per = round(country_app_data['installs'].isnull().mean()* 100.00,2)
min_install_per = round(country_app_data['min_installs'].isnull().mean()* 100.00,2)
price_per = round(country_app_data['price'].isnull().mean()* 100.00,2)

score_per = round(country_app_data['score'].isnull().mean()* 100.00,2)

print(f'Installs null percentage: {install_per}%')
print(f'Min Installs null percentage: {min_install_per}%')
print(f'price null percentage: {price_per}%')

print(f'\nTotal percentage of Install, min_Installs, price :{install_per + min_install_per + price_per}')

print(f'\nScore null percentage: {score_per}%')

Installs null percentage: 0.29%
Min Installs null percentage: 0.29%
price null percentage: 0.29%

Total percentage of Install, min_Installs, price :0.8699999999999999

Score null percentage: 33.69%


1. Rows with missing install and price information represented less than 0.3% of the dataset and were removed to preserve data integrity without materially affecting the analysis.

In [137]:
country_app_data.dropna(subset = ['installs','min_installs','price'], inplace = True)

2. score, ratings, reviews (37,496 missing values)

These are missing for about 33.7% of the data, so dropping them would remove one-third of your dataset.

> Keep NaN for score, ratings, and reviews because they represent unavailable information rather than zero.

- fillna(0) - If we observe data in those columns they have 0 as data. Filling with 0 will mislead that from unavailable information to 0 rating, score or review.

- fillna('median()') - Score is calculated by average of ratings. Since ratings is also have null values, calculating average of ratings with null values and takinng median is not correct information.

In [138]:
country_app_data.duplicated().sum()

np.int64(1)

In [139]:
country_app_data.drop_duplicates(inplace = True)

In [140]:
country_app_data.duplicated().sum()

np.int64(0)

In [141]:
country_app_data.shape

(110987, 9)

In [142]:
country_app_data.to_csv('country_apps.csv', index = False)

## **Data Cleaning of Global Apps**

In [143]:
global_app_data = pd.read_csv('/content/apps-main.csv')

In [144]:
global_app_data.head()

,app_id,title,description,summary,installs,min_installs,max_installs,score,ratings,reviews,price,free,developer,genre,ad_supported,in_app_purchases,size,updated
0,com.droid4you.application.wallet,Wallet: Budget Expense Tracker,<b>Wallet is the all-in-one personal finance m...,"All-in-one personal finance app: Budget, track...","10,000,000+",10000000,12778093,4.602770,376251.0,7477.0,0.0,True,BudgetBakers,Finance,False,True,NaN,1.776860e+09
1,com.freeman.moneymanager,Money Tracker-Expense & Budget,Save money effortlessly! \r\n\r\nMoney Tracker...,"Money management made simple. Money manager, c...","5,000,000+",5000000,8112001,4.868132,284764.0,921.0,0.0,True,Horoscope365,Finance,True,True,NaN,1.776151e+09
2,com.picsart.studio,"Picsart AI Photo Editor, Video",Ignite your creative potential with Picsart — ...,Remove and swap backgrounds in your photos and...,"1,000,000,000+",1000000000,1404716134,4.032339,12207068.0,445242.0,0.0,True,"PicsArt, Inc.",Photography,True,True,NaN,1.777555e+09
3,com.colpit.diamondcoming.isavemoney,Budget planner—Expense tracker,Budget Planner on the go!\r\n<h1>Your Ultimate...,Master money management with Budget Planner: E...,"5,000,000+",5000000,5585324,4.271907,51002.0,3147.0,0.0,True,Battery Stats Saver,Finance,True,True,NaN,1.754659e+09
4,com.photoroom.app,Photoroom: AI Photo Editor,Create pro images that sell with Photoroom AI....,"Remove backgrounds, edit and retouch photos fo...","100,000,000+",100000000,173236105,4.684298,3881733.0,17581.0,0.0,True,Photoroom AI Photo Editor,Photography,False,True,NaN,1.777974e+09


In [145]:
global_app_data.shape

(690, 18)

In [146]:
global_app_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 690 entries, 0 to 689
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   app_id            690 non-null    object 
 1   title             690 non-null    object 
 2   description       690 non-null    object 
 3   summary           690 non-null    object 
 4   installs          690 non-null    object 
 5   min_installs      690 non-null    int64  
 6   max_installs      690 non-null    int64  
 7   score             686 non-null    float64
 8   ratings           686 non-null    float64
 9   reviews           686 non-null    float64
 10  price             690 non-null    float64
 11  free              690 non-null    bool   
 12  developer         690 non-null    object 
 13  genre             690 non-null    object 
 14  ad_supported      690 non-null    bool   
 15  in_app_purchases  690 non-null    bool   
 16  size              0 non-null      float64
 1

In [147]:
global_app_data['size'].unique()

array([nan])

In [148]:
unwanted_col = ['description', 'summary', 'size']

global_app_data.drop(columns= unwanted_col, inplace = True)

In [149]:
global_app_data.head()

,app_id,title,installs,min_installs,max_installs,score,ratings,reviews,price,free,developer,genre,ad_supported,in_app_purchases,updated
0,com.droid4you.application.wallet,Wallet: Budget Expense Tracker,"10,000,000+",10000000,12778093,4.602770,376251.0,7477.0,0.0,True,BudgetBakers,Finance,False,True,1.776860e+09
1,com.freeman.moneymanager,Money Tracker-Expense & Budget,"5,000,000+",5000000,8112001,4.868132,284764.0,921.0,0.0,True,Horoscope365,Finance,True,True,1.776151e+09
2,com.picsart.studio,"Picsart AI Photo Editor, Video","1,000,000,000+",1000000000,1404716134,4.032339,12207068.0,445242.0,0.0,True,"PicsArt, Inc.",Photography,True,True,1.777555e+09
3,com.colpit.diamondcoming.isavemoney,Budget planner—Expense tracker,"5,000,000+",5000000,5585324,4.271907,51002.0,3147.0,0.0,True,Battery Stats Saver,Finance,True,True,1.754659e+09
4,com.photoroom.app,Photoroom: AI Photo Editor,"100,000,000+",100000000,173236105,4.684298,3881733.0,17581.0,0.0,True,Photoroom AI Photo Editor,Photography,False,True,1.777974e+09


In [150]:
global_app_data.columns

Index(['app_id', 'title', 'installs', 'min_installs', 'max_installs', 'score',
       'ratings', 'reviews', 'price', 'free', 'developer', 'genre',
       'ad_supported', 'in_app_purchases', 'updated'],
      dtype='object')

In [151]:
global_app_data['installs'] = pd.to_numeric(global_app_data['installs']
                               .astype(str)
                               .str.replace(',', '', regex = False)
                               .str.replace('+', '', regex = False),
                               errors = 'coerce')

In [152]:
global_app_data['genre'].unique()

array(['Finance', 'Photography', 'Productivity', 'Entertainment',
       'Education', 'Art & Design', 'Business', 'Tools', 'Lifestyle',
       'Health & Fitness', 'Casual', 'Medical', 'Social',
       'Travel & Local', 'Shopping', 'Video Players & Editors',
       'Food & Drink', 'Music & Audio', 'Adventure', 'Personalization',
       'Action', 'Communication', 'Educational', 'Strategy',
       'Role Playing', 'Maps & Navigation', 'Sports', 'Trivia', 'Puzzle',
       'Arcade', 'Simulation', 'Racing', 'Board'], dtype=object)

In [153]:
global_app_data['genre'].replace({"Educational" : "Education"}, inplace= True)

/tmp/ipykernel_2878/2763082364.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  global_app_data['genre'].replace({"Educational" : "Education"}, inplace= True)


In [154]:
global_app_data['genre'].unique()

array(['Finance', 'Photography', 'Productivity', 'Entertainment',
       'Education', 'Art & Design', 'Business', 'Tools', 'Lifestyle',
       'Health & Fitness', 'Casual', 'Medical', 'Social',
       'Travel & Local', 'Shopping', 'Video Players & Editors',
       'Food & Drink', 'Music & Audio', 'Adventure', 'Personalization',
       'Action', 'Communication', 'Strategy', 'Role Playing',
       'Maps & Navigation', 'Sports', 'Trivia', 'Puzzle', 'Arcade',
       'Simulation', 'Racing', 'Board'], dtype=object)

In [155]:
global_app_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 690 entries, 0 to 689
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   app_id            690 non-null    object 
 1   title             690 non-null    object 
 2   installs          690 non-null    int64  
 3   min_installs      690 non-null    int64  
 4   max_installs      690 non-null    int64  
 5   score             686 non-null    float64
 6   ratings           686 non-null    float64
 7   reviews           686 non-null    float64
 8   price             690 non-null    float64
 9   free              690 non-null    bool   
 10  developer         690 non-null    object 
 11  genre             690 non-null    object 
 12  ad_supported      690 non-null    bool   
 13  in_app_purchases  690 non-null    bool   
 14  updated           672 non-null    float64
dtypes: bool(3), float64(5), int64(3), object(4)
memory usage: 66.8+ KB


In [156]:
global_app_data.isnull().sum()

,0
app_id,0
title,0
installs,0
min_installs,0
max_installs,0
score,4
ratings,4
reviews,4
price,0
free,0


In [157]:
global_app_data.dropna(subset = ['installs','min_installs','price'], inplace= True)

In [158]:
global_app_data.isnull().sum()

,0
app_id,0
title,0
installs,0
min_installs,0
max_installs,0
score,4
ratings,4
reviews,4
price,0
free,0


In [159]:
global_app_data.duplicated().sum()

np.int64(0)

In [160]:
global_app_data.shape

(690, 15)

In [161]:
global_app_data.to_csv('global_apps.csv', index = False)